references/cli/adopt-templates/default.cue

```diff
@@ -89,101 +90,93 @@ import (
 		_category: *"unknown" | string
 		for key, kinds in resourceCategoryMap if list.Contains(kinds, r.kind) {
 			_category: key
-		},
+		}
+		_cluster: *"local" | string
 		if r.metadata.annotations != _|_ if r.metadata.annotations["app.oam.dev/cluster"] != _|_ {
 			_cluster: r.metadata.annotations["app.oam.dev/cluster"]
 		}
 	}]
-	_clusters: [ for r in _resources if r._cluster != _|_ {r._cluster} ]
-	resourceMap: {
-		for key, val in resourceCategoryMap {
-			"\(key)": [ for r in _resources if r._category == key {r}]
-		}
-		unknown: [ for r in _resources if r._category == "unknown" {r}]
-	}
 
-	unknownKinds: {for r in resourceMap.unknown {"\(r.kind)": true}}
-	unknownByKinds: {for kind, val in unknownKinds {
-		"\(kind)": [ for r in resourceMap.unknown if r.kind == kind {r}]
-	}}
+	_clusters: [ for _cluster, _ in {for r in _resources {"\(r._cluster)": true}} {_cluster}]
 
 	appName: $args.appName
-	comps: [
-		if len(resourceMap.crd) > 0 {
-			type: "k8s-objects"
-			name: "crds"
-			properties: objects: [ for r in resourceMap.crd {
-				apiVersion: r.apiVersion
-				kind:       r.kind
-				metadata: name: r.metadata.name
-				if r._cluster != _|_ {
-					metadata: annotations: "app.oam.dev/cluster": (r._cluster)
-				}
-			}]
-		},
-		if len(resourceMap.ns) > 0 {
-			type: "k8s-objects"
-			name: "ns"
-			properties: objects: [ for r in resourceMap.ns {
-				apiVersion: r.apiVersion
-				kind:       r.kind
-				metadata: name: r.metadata.name
-				if r._cluster != _|_ {
-					metadata: annotations: "app.oam.dev/cluster": (r._cluster)
-				}
-			}]
-		},
-		for r in resourceMap.workload + resourceMap.service {
-			type: "k8s-objects"
-			name: strings.ToLower("\(r.kind)-\(r.metadata.name)")
-			properties: objects: [{
-				apiVersion: r.apiVersion
-				kind:       r.kind
-				metadata: name:      r.metadata.name
-				if r.metadata.namespace != _|_ {
-					metadata: namespace: r.metadata.namespace
-				}
-				spec: r.spec
-			}]
-		},
-		for key in ["config", "sa", "operator", "storage"] if len(resourceMap[key]) > 0 {
-			type: "k8s-objects"
-			name: "\(key)"
-			properties: objects: [ for r in resourceMap[key] {
-				apiVersion: r.apiVersion
-				kind:       r.kind
-				metadata: name: r.metadata.name
-				if r.metadata.namespace != _|_ {
-					metadata: namespace: r.metadata.namespace
-				}
-				if r._cluster != _|_ {
-					metadata: annotations: "app.oam.dev/cluster": (r._cluster)
+	clusterEntries: {
+		for cluster in _clusters {
+			"\(cluster)": {
+				_prefix: *"" | string
+				if cluster != "local" {
+					_prefix: cluster + ":"
```

Comment:
it's better to add prefix like this, cluster+":"+namespace+":", because there are some resources like deploy and service may have same name and cluster.

Reply:
Yes. The design here is that this "_prefix" is added to the component name instead of the resource name. So for multiple deployments within one cluster, they will be put in one component together. It only has logical effects on the application, and will not affect the dispatched resources.